app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,45 @@
+class CommentsController < ApplicationController
+  before_action :require_comment, only: [:destroy, :update]
+
+  def create
+    if Comment.create(comment_params)
```

Comment:
Šeit situacija ir lidziga kā comunities ar posts un tev tomēr naksies definet vel vienu nested routes limēni.

Bet tagad, kad tev ir pieredze ar tiem, tu vari izmantot shallow nesting
https://edgeguides.rubyonrails.org/routing.html#shallow-nesting
Izskatisies kau kā tā:

```ruby
  resources :communities do
    resources :posts, shallow: true do
      resources :comments, shallow: true, except: [:new, :edit]
    end
  end 
```

Doma ir tada, kā sanak diezgan slikta situacija kād komentaru nevar izveidot. Komentaram taču ir validacija uz content un pieņemsim kad lietotais raksta sacerejumu viņu mēt uz BACK (par to vel atseviška sarūna) un viņš pazaudes visu saturu no formas.

Tev tomēr vajag izmantot @post instanci iekš kontroleri un vel vajag instanci priekš @comment. Sanaks neliels refaktorings. Bet iekš kontroleri tas izskatisies aptuveni tā:

```ruby
def create
  @post = Post.find_by(params[:post_id])
  @comment = Comment.new(comment_params)
  
  if @comment.save
    redirect_to show_post(@post)
  else
   render 'posts/show'
  end
end
```

Kā tu redzi augstak, jā būs kļuda, vnk ierenderesim posts/show lapu.


Un tad iekš posts_controller tev vienmer būs tira jauna instance, lai posts/show varetu izmantot gan tur, gan šeit

```ruby
def show
  @comment = Comment.new
  ...
end
```

Reply:
@ErnestsEida @pavels-gaimovics glūži nenolasu Pāvela komentārā, ka šī rindiņa vienmēr ir true -> `if Comment.create(comment_params)`, līdz ar to if ir bezjēdzīgs un šajā situācijā arī kļūdains.

@ErnestsEida izlasi iekš dokumentācias un iztestē konsolē, ka create metodes vienmēr atgriež ActiveRecord instani  un tā var būt vai nu valīdas un saglabāta vai arī nevalīda un nesaglabāta, bet `if` vienmēr ir `true`, jo Comment.new nav falsly value. Tādas ir tikai `false` un `nil`.

Līdz ar to, šadā risinājumā tu nekad neienāc iekš else zara!

Reply:
@pavels-gaimovics @ErnestsEida Pāvels jau demonstrē super high level nesting, bet es parasti pieturos pie 2viem līmeņiem.

Nekas neaizliedz izveidot arī šādu route config:

```ruby
resources :communities do
  resources :posts
end

resources :posts, only: [] do
  resources :comments
end
```

bet šis ir kinda hacks tam shallow risinājumam, lai route helperi nebūtu pārāk long.

Tomēr es paredzu ka nākotnē var lieti noderēt iespēja, ka piemēram post var atvērt arī ne caur nested communities - lai īsāks url.

Reply:
@pavels-gaimovics pie comment create piedāvātās post ielases notiekti nevajadzētu izmantot Post.find_by.

Postam ir jabūt vienmēr - komentāru nevar nesaglabāt, ja nav post, tāpēc nav jēgas laist tālāk, ja post_id nav.